### PR TITLE
Add Broadcaster.StartLogging for scheduler

### PR DESCRIPTION
**Pull request by ping035627**
**What type of PR is this?**
 /kind cleanup

**What this PR does / why we need it**:
Add Broadcaster.StartLogging for scheduler.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

/release-note-none
/priority backlog

### CODE REVIEW

**Comment by misterikkit** [comment]
Hi @ping035627, could you explain a bit about this PR? I'm not super familiar with event broadcaster. Is `StartLogging` something every control plane component should do? Is there an issue or some documentation to reference?

**Comment by ping035627** [comment]
@misterikkit, I think we should print the result of scheduling like kubelet or contorller-manager. And I have searched the issue, it has been discussed, #67589, but the problem has been shelved. I hope it can be review. Thanks.

**Comment by misterikkit** [comment]
> I think we should print the result of scheduling like kubelet or contorller-manager.

I completely agree with this. I think I just don't fully understand the mechanism. Is this taking events which the scheduler sends to API server and echoing them to the scheduler log file?

Also, thanks for pointing me to that issue. I was not aware of it when I filed https://github.com/kubernetes/kubernetes/issues/71819. That issue has a PR referenced and I wonder how much overlap there is between the two efforts.

**Comment by ping035627** [comment]
> I completely agree with this. I think I just don't fully understand the mechanism. Is this taking events which the scheduler sends to API server and echoing them to the scheduler log file?
This is related to the event mechanism, Broadcaster.StartLogging is just watch the event of EventBroadcaster, and then print it.

>  I was not aware of it when I filed #71819. That issue has a PR referenced and I wonder how much overlap there is between the two efforts.
About the issue #71819, Broadcaster.StartLogging can implement part of functions you want, it will print log like follows:
`I1214 11:42:12.366399    3104 event.go:221] Event(v1.ObjectReference{Kind:"Endpoints", Namespace:"kube-system", Name:"kube-scheduler", UID:"3dc62d05-ff52-11e8-84ec-525400a205f1", APIVersion:"v1", ResourceVersion:"238", FieldPath:""}): type: 'Normal' reason: 'LeaderElection' k8sMaster_3d2b0982-ff52-11e8-9655-525400a205f1 became leader
I1214 11:44:56.702387    3104 event.go:221] Event(v1.ObjectReference{Kind:"Pod", Namespace:"kube-system", Name:"calico-node-qn8nn", UID:"9fb47dec-ff52-11e8-84ec-525400a205f1", APIVersion:"v1", ResourceVersion:"373", FieldPath:""}): type: 'Normal' reason: **'Scheduled' Successfully assigned kube-system/calico-node-qn8nn to 10.46.179.22**
I1214 11:44:56.705723    3104 event.go:221] Event(v1.ObjectReference{Kind:"Pod", Namespace:"kube-system", Name:"calico-kube-controllers-666d8855c9-st8v7", UID:"9fb4b4a6-ff52-11e8-84ec-525400a205f1", APIVersion:"v1", ResourceVersion:"374", FieldPath:""}): type: 'Warning' reason: **'FailedScheduling' 0/2 nodes are available: 2 node(s) had taints that the pod didn't tolerate.**`

File: cmd/kube-scheduler/app/server.go [hunk 1]
```diff
@@ -197,6 +197,7 @@ func Run(cc schedulerserverconfig.CompletedConfig, stopCh <-chan struct{}) error
 
 	// Prepare the event broadcaster.
 	if cc.Broadcaster != nil && cc.EventClient != nil {
+		cc.Broadcaster.StartLogging(klog.V(3).Infof)
```
Comment (resouer) [inline cmd/kube-scheduler/app/server.go]: I don't have strong objection on this idea, but `V(3)` seems too much verbose

**Comment by ping035627** [comment]
@bsalamat, What‘s your opinions?

File: cmd/kube-scheduler/app/server.go (hunk 1 above)
Comment (bsalamat) [reply cmd/kube-scheduler/app/server.go]: I agree with @resouer. We should probably go with V(6).

Comment (ping035627) [reply cmd/kube-scheduler/app/server.go]: @bsalamat, Okay, I changed it to V(6), need some label, thanks.

**Review by bsalamat — COMMENTED** [review]
/lgtm
/approve

**Comment by misterikkit** [comment]
I'm not sure if the integration test failures are flakes or not. @ping035627 are you able to investigate?

**Comment by ping035627** [comment]
@misterikkit In my CI environment, integration tests failures sometimes occur, and each failure has different use cases. Considering that multiple test cases under packages may interact with each other, single package execution is generally normal, we are also looking into this problem.

**Pull request merged**